feat/feature.go

```
@@ -105,6 +105,45 @@ type Collection interface {
 	Location() Feature
 }
 
+// BasePositionOf returns the position in f converted to coordinates in the
+// deepest non nil feature location and the reference feature which may be the
+// feature itself if it has a nil Location.
+// BasePositionOf will panic if the feature chain is deeper than 1000 links.
+func BasePositionOf(f Feature, position int) (int, Feature) {
+	if f == nil {
```

Should I also let this panic? I think that the return kind of makes sense here.